podcast/2.17.md

```diff
@@ -0,0 +1,487 @@
+
+And so there were all these cool things you could do to code effectively. All these Pivots knew all these tricks to get code fast and make changes fast and I started picking those up. But then there was a certain class of projects at Pivotal where the IDEs... There were codebases that were too large to be opened in IDEs or codebases in different languages. And so there were other Pivots who had all this deep expertise in VIM and using all these lower level tools for coding.
+
+**Beyang Liu:**
+
+Interesting. So they would use VIM on the larger codebases, because you would try to open it up in the IDE and it would slow to a crawl or something like that.
+
+**Max Brunsfeld:**
+
+Yeah. At that time, RubyMine couldn't open–I don't know what would happen–some of these client codebases in the IDEs. There's this whole other parallel knowledge base at Pivotal of people who knew how to do crazy VIM tricks.
+
+**Beyang Liu:**
+
+Like the dark art of VIM.
+
+**Max Brunsfeld:**
+
+Right. And so I learned all of that as well. That was cool for me because I was interested in learning other programming languages too. I had learned a few on the job, but at the time I was interested in Clojure and Haskell and learning functional languages and there weren't IDEs for those. So I was really happy to be learning how to use tools like VIM. But while learning those, I remember it was so much work to customize them and make them usable. Adding modern features to them was difficult and so was learning all the commands that you needed to know to make changes inside brackets and using all these commands to edit pieces of text.
+
+It's funny, Nathan and I had had all these conversations while working on this side project. Something else that Nathan was really interested in was like, he was like, “One day I want to build my own code editor that's going to be better than all these code editors. And one thing about it, it's going to be the IDEs with all these syntax aware commands, but there's going to be incremental parsing so that it's fast. It's not slow like these IDEs.”
+
+And that was the extent of it, that was what I took away from those conversations. But I remember putting this idea into my head about, "Oh, incremental parsing that makes sense that you want to do, but that isn't a Salt." And then I could see that once I switched to using VIM, I could understand, like, oh, VIM doesn't really do syntax analysis. There's this really rough syntax highlighting and then the rest is all from this power of image based on text processing, not really code processing. So it made sense to me then, once I switched back from using IDEs to using VIM.
```

Comment:
I don't understand a couple of bits here, so maybe worth revisiting the recording. What is Salt here? And what is "power of image based on text processing"?